application/app/Models/InstitutionUser.php

```diff
@@ -164,4 +170,20 @@ protected function deactivationDate(): Attribute
             }
         );
     }
+
+    public function isOnlyUserWithRootRole(): bool
+    {
+        $rootRole = $this->roles()->where('is_root', true)->with('institutionUsers')->first();
+        $rootRoleInstitutionUserIds = collect($rootRole->institutionUsers)->pluck('id');
+
+        return $rootRoleInstitutionUserIds->count() < 2 && $rootRoleInstitutionUserIds->contains($this->id);
+    }
```

Comment:
**[non-blocking]** I also found that `isOnlyUserWithRootRole()` and `hasRootRole()` are used only together so maybe it makes sense to merge them into one function `hasRootRoleAndIsOnlyOneWithIt()`. Suggested naming is not the best option 😄

Reply:
I left them as separate since I had a feeling that it may be important to check in some cases if user just has root role, but yea, I'm not sure if it is even needed :smile: